shell/browser/net/resolve_host_helper.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018 GitHub, Inc.
```

Comment:
This is based on resolve_proxy_helper.cc and `extensions/browser/api/dns/dns_api.cc` from Chromium.